### ISSUE TITLE
RealZopeInterfaceTests fails on z.i versions containing non-digits

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6007]
|<img alt="ivank's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| ivank reported|
|-|-|
|Trac ID|trac#6007|
|Type|defect|
|Created|2012-09-20 03:29:35Z|

zope.interface versions can have letters (see http://old.zope.org/Products/ZopeInterface/ or install from SVN to get something like "zope.interface-3.6.2dev-py2.7.egg-info"), but `RealZopeInterfaceTests.install` tries to `int()` all parts of the version string, resulting in test failure:

```
[ERROR]
Traceback (most recent call last):
  File "C:\Python27-32\lib\unittest\case.py", line 318, in run
    testMethod()
  File "c:\users\root\documents\projects\twisted\twisted\test\test_twisted.py", line 515, in test_zope35
    self.install((3, 5))
  File "c:\users\root\documents\projects\twisted\twisted\test\test_twisted.py", line 633, in install
    versionTuple = tuple(int(part) for part in installed.split('.'))
  File "c:\users\root\documents\projects\twisted\twisted\test\test_twisted.py", line 633, in <genexpr>
    versionTuple = tuple(int(part) for part in installed.split('.'))
exceptions.ValueError: invalid literal for int() with base 10: '2dev'

twisted.test.test_twisted.RealZopeInterfaceTests.test_zope35
===============================================================================
[ERROR]
Traceback (most recent call last):
  File "C:\Python27-32\lib\unittest\case.py", line 318, in run
    testMethod()
  File "c:\users\root\documents\projects\twisted\twisted\test\test_twisted.py", line 536, in test_zope36
    self.install((3, 6))
  File "c:\users\root\documents\projects\twisted\twisted\test\test_twisted.py", line 633, in install
    versionTuple = tuple(int(part) for part in installed.split('.'))
  File "c:\users\root\documents\projects\twisted\twisted\test\test_twisted.py", line 633, in <genexpr>
    versionTuple = tuple(int(part) for part in installed.split('.'))
exceptions.ValueError: invalid literal for int() with base 10: '2dev'

twisted.test.test_twisted.RealZopeInterfaceTests.test_zope36
===============================================================================
```

Attachments:

* [6007-01-int-fewer-parts.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/5a8/5a8c29fa727e5182a4a84bcd2708df748bb7e117/9857a8c347017333b9cbd5333f2dcb9e9545fcb0.patch) (895 bytes) - added by ivank on 2012-09-20 03:32:17Z - 
* [6007-02-int-fewer-parts.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/5a8/5a8c29fa727e5182a4a84bcd2708df748bb7e117/dbb8b61e1b6bfdf5541122db5dad3f22c2593dae.patch) (913 bytes) - added by ivank on 2012-09-20 03:35:45Z - fixed column limit

<details><summary>Searchable metadata</summary>

```
trac-id__6007 6007
type__defect defect
reporter__ivank ivank
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__easy easy
time__1348111775000000 1348111775000000
changetime__1348137425000000 1348137425000000
version__None None
owner__exarkun exarkun
cc__ivank
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="ivank's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_1"></a>|ivank commented|
|-|-|

`test_zope35` and `test_zope36` run properly here when patched.  Please review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

Thanks.  Looks good.  I'll apply (with a misc fragment).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set status to `closed`|
|-|-|

(In [35746]) Improve robustness of test for expected zope.interface behavior

Author: ivank
Reviewer: exarkun
Fixes: [#6007](https://github.com/twisted/twisted/issues/6007)

Account for possible zope.interface versions like "3.6.2dev".
